internal/api/mfa.go

```diff
@@ -676,6 +874,115 @@ func (a *API) verifyPhoneFactor(w http.ResponseWriter, r *http.Request, params *
 	return sendJSON(w, http.StatusOK, token)
 }
 
+func (a *API) verifyWebAuthnFactor(w http.ResponseWriter, r *http.Request, params *VerifyFactorParams) error {
+	ctx := r.Context()
+	user := getUser(ctx)
+	factor := getFactor(ctx)
+	db := a.db.WithContext(ctx)
+
+	var webAuthn *webauthn.WebAuthn
+	var credential *webauthn.Credential
+	var err error
+
+	switch {
+	case params.WebAuthn == nil:
+		return badRequestError(ErrorCodeValidationFailed, "WebAuthn config required")
+	case factor.IsVerified() && params.WebAuthn.AssertionResponse == nil:
+		return badRequestError(ErrorCodeValidationFailed, "WebAuthn Assertion Response required to login")
+	case factor.IsUnverified() && params.WebAuthn.CreationResponse == nil:
+		return badRequestError(ErrorCodeValidationFailed, "WebAuthn Creation Response required to login")
```

Comment:
Maybe use the JSON field names in these error messages? Something like `credential_creation_options missing` etc...